test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Import/ImportTests.cs

```diff
@@ -183,14 +183,119 @@ public async Task GivenIncrementalLoad_80KSurrogateIds_BadRequestIsReturned()
         }
 
         [Fact]
-        public async Task GivenIncrementalLoad_MultipleInputVersionsOutOfOrderSomeNotExplicit_ResourceNotExisting_NoGap()
+        public async Task GivenIncrementalLoad_MultipleInputsWithImplicitVersionsExplicitLastUpdatedBeforeImplicit()
+        {
+            var id = Guid.NewGuid().ToString("N");
+            var ndJson1 = CreateTestPatient(id);
+            var ndJson2 = CreateTestPatient(id, DateTimeOffset.UtcNow.AddDays(-1));
+            var location = (await ImportTestHelper.UploadFileAsync(ndJson1 + ndJson2, _fixture.StorageAccount)).location;
+            var request = CreateImportRequest(location, ImportMode.IncrementalLoad, false, true);
+            await ImportCheckAsync(request, null, 0);
+
+            var history = await _client.SearchAsync($"Patient/{id}/_history");
+            Assert.Single(history.Resource.Entry);
+            Assert.Equal("1", history.Resource.Entry[0].Resource.VersionId);
```

Comment:
I think I'm missing something here. Why aren't both versions included?

Reply:
Both resources have the same content. Version with explicit last updated is loaded first. When we attempt to load version with implicit last updated we check current database state. We see that there is version with identical content, and we return - loaded. Hence, both input records were loaded :-), but only one version is actually present in the database.